app/workers/functions.php

```diff
@@ -150,21 +150,25 @@ function: $function,
                     throw new Exception('Function not found (' . $function->getId() . ')');
                 }
 
-                if ($scheduleOriginal && $scheduleOriginal !== $function->getAttribute('schedule')) { // Schedule has changed from previous run, ignore this run.
+                if ($functionOriginal->getAttribute('schedule') !== $function->getAttribute('schedule')) { // Schedule has changed from previous run, ignore this run.
+                    return;
+                }
+
+                if ($functionOriginal->getAttribute('scheduleUpdatedAt') !== $function->getAttribute('scheduleUpdatedAt')) { // Double execution due to rapid cron changes, ignore this run.
```

Comment:
Instead of introducing a new attribute, can we make use of scheduledNext? If there's a duplicate execution scheduled, one would execute 1st and update the scheduledNext. When the 2nd one executes and fetches the function from the DB, there would be a different scheduledNext than from the event. What do you think?

Reply:
Nice idea! I will attempt this and see if it works as expected. Sounds logical.

Reply:
Gave it some more thinking, and I would be more confident with separate attribute. That way, the attribute is directly linked to corn itself.

The problem I can see is that if you scale and have multiple workers, it will result in multiple parallel jobs. That means both jobs would be taken at the same time, so they both pass the check. 

Also, I know Torsten plans to work on a new queuing library that would solve this with jobIds. I don't see new attribute as too big of a concern, since it will be removed and replaced with whole different attribute in future - no data migration needed.

Reply:
Okay sure. Let's at least add a comment about why the `scheduleUpdatedAt` attribute is needed (maybe in `app/config/collections.php`) and that it can be removed once jobIds come.

Reply:
Added comment into collections.php